Framework/CurveFitting/src/FuncMinimizers/FABADAMinimizer.cpp

```diff
@@ -976,53 +650,482 @@ void FABADAMinimizer::SimAnnealingRefrigeration() {
+}
+
+/** Create the workspace containing the converged chain
+*
+* @param convLength :: length of the converged chain
+* @param nSteps :: number of steps done between chain points to avoid
+*correlation
+*/
+void FABADAMinimizer::outputConvergedChains(size_t convLength, int nSteps) {
+
+  // Create the workspace for the converged part of the chain.
+  API::MatrixWorkspace_sptr wsConv;
+  if (convLength > 0) {
+    wsConv = API::WorkspaceFactory::Instance().create(
+        "Workspace2D", m_nParams + 1, convLength, convLength);
+  } else {
+    g_log.warning() << "Empty converged chain, empty Workspace returned.";
+    wsConv = API::WorkspaceFactory::Instance().create("Workspace2D",
+                                                      m_nParams + 1, 1, 1);
+  }
+
+  // Do one iteration for each parameter plus one for Chi square.
+  for (size_t j = 0; j < m_nParams + 1; ++j) {
+    std::vector<double>::const_iterator first =
+        m_chain[j].begin() + m_convPoint;
+    std::vector<double>::const_iterator last = m_chain[j].end();
+    std::vector<double> conv_chain(first, last);
+    auto &X = wsConv->mutableX(j);
+    auto &Y = wsConv->mutableY(j);
+    for (size_t k = 0; k < convLength; ++k) {
+      X[k] = double(k);
+      Y[k] = conv_chain[nSteps * k];
+    }
+  }
+
+  // Set and name the workspace for the converged part of the chain.
+  setProperty("ConvergedChain", wsConv);
+}
+
+/** Create the workspace containing chi2 values
+*
+* @param convLength :: length of the converged chain
+* @param mostProbableChi2 :: most probable chi2 value
+*correlation
+*/
+void FABADAMinimizer::outputCostFunctionTable(size_t convLength,
+                                              double mostProbableChi2) {
+  // Create the workspace for the Chi square values.
+  API::ITableWorkspace_sptr wsChi2 =
+      API::WorkspaceFactory::Instance().createTable("TableWorkspace");
+  wsChi2->addColumn("double", "Chi2min");
+  wsChi2->addColumn("double", "Chi2MP");
+  wsChi2->addColumn("double", "Chi2min_red");
+  wsChi2->addColumn("double", "Chi2MP_red");
+
+  // Obtain the quantity of the initial data.
+  API::FunctionDomain_sptr domain = m_leastSquares->getDomain();
+  size_t data_number = domain->size();
+
+  // Calculate the value for the reduced Chi square.
+  double Chi2min_red =
+      m_chi2 / (double(data_number - m_nParams)); // For de minimum value.
+  double mostPchi2_red;
+  if (convLength > 0)
+    mostPchi2_red = mostProbableChi2 / (double(data_number - m_nParams));
+  else {
+    g_log.warning()
+        << "Most probable chi square not calculated. -1 is returned.\n"
+        << "Most probable reduced chi square not calculated. -1 is "
+           "returned.\n";
+    mostPchi2_red = -1;
+  }
+
+  // Add the information to the workspace and name it.
+  API::TableRow row = wsChi2->appendRow();
+  row << m_chi2 << mostProbableChi2 << Chi2min_red << mostPchi2_red;
+  setProperty("CostFunctionTable", wsChi2);
+}
+
+/** Create the workspace containing PDF
```

Comment:
replace PDF with 'probability distribution function (PDF)' so it is clear what PDF stands for.